### PR TITLE
#11788 Pin tox to latest version 3 for GitHub Actions.

### LINKED ISSUE #11788: GitHub Actions CI broken due to latest tox
**Describe the incorrect behavior you saw**
```
ROOT: will run in automatically provisioned tox, host /opt/hostedtoolcache/Python/3.10.8/x64/bin/python is missing [requires (has)]: tox-wheel>=0.6.0
ROOT: provision> .tox/.tox/bin/python -m tox
/home/runner/work/twisted/twisted/.tox/.tox/bin/python: No module named tox
Error: Process completed with exit code 1.
```

**Describe how to cause this behavior**

Just run any test on our CI.

See a build run here 

https://github.com/twisted/twisted/actions/runs/3574757910/jobs/6212002907#step:10:12

**Additional context**

I guess that this is due to tox 4.

I previous run of the same commit, was green and was using "Using cached tox-3.27.1-py2.py3-none-any.whl (86 kB)"

The failing CI using "Downloading tox-4.0.8-py3-none-any.whl"

### PR DESCRIPTION
**Pull request by adiroiban**
## Scope and purpose

Fixes #11788

This tries to get our CI green again with the least effort.

In a separate PR we can look at removing our dependence on tox.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -20,6 +20,7 @@ minversion=3.24.1
 requires=
     virtualenv>=20.7.2
     tox-wheel>=0.6.0
+    tox < 4
```
Comment (adiroiban) [inline tox.ini:23]: Even if you run tox v3, when it creates ~/.tox/.tox venv, it will create it with tox v4.

This looks like it can fix it.

**Comment by adiroiban** [comment]
This is ready for review. Thanks!

needs-review

**Review by glyph — APPROVED** [review]
We should really develop a comprehensive strategy for pinning *everything*, and not just with inequalities; we should use exact versions everywhere (except for our package metadata) and let dependabot send upgrade PRs.

But, let's get trunk back to passing ASAP.

**Comment by glyph** [comment]
> In a separate PR we can look at removing our dependence on tox.

What would you suggest replacing it with? `nox`?

**Pull request merged**